src/app/components/Header/HeaderSearchBar/HeaderSearchBar.test.tsx

```diff
@@ -144,4 +144,23 @@ describe("HeaderSearchBar", () => {
 
 		expect(onSearch).toHaveBeenCalledWith("test");
 	});
+
+	it("should always show clear button", async () => {
+		render(<HeaderSearchBar alwaysDisplayClearButton />);
+
+		await userEvent.click(screen.getByRole("button"));
+
+		expect(screen.getByTestId("header-search-bar__reset")).toBeInTheDocument();
+		expect(screen.getByTestId("header-search-bar__reset-icon")).toHaveClass("w-4");
```

Comment:
I think having `expect(screen.getByTestId("header-searchbar__reset")).toBeInTheDocument();`  is sufficient, having class name checks would force us to fix tests even we do small CSS changes, what do you think?

Reply:
Oh I added this bc of line coverage, since the classes are being added conditionally based on the presence of this new prop